engine/api/src/main/java/io/deephaven/engine/table/Table.java

```diff
@@ -355,6 +355,11 @@ public interface Table extends
      * <p>
      * If the firstPosition is negative and the lastPosition is negative, they are both counted from the end of the
      * table. For example, slice(-2, -1) returns the second to last row of the table.
+     * <p>
+     * If firstPosition is negative and lastPosition is positive, then firstPosition is counted from the end of the
+     * table, inclusively. The lastPosition is counted from the beginning of the table. For example, slice(-3, 5)
+     * returns all rows starting from the third-last row to the fourth row of the table. If there are no rows between
```

Comment:
Am I thinking wrong or is the 6th row of the table selected for index 5?

Reply:
You are not wrong. Both firstPosition and lastPosition are indexes starting from 0. And the subset of rows returned are in the indexes [firstPosition, lastPosition).  So instead of fourth, I should have written fifth in the comment. The sixth row is excluded. I will update this.